### PR TITLE
be able to skip automatic closing of a hash resource

### PR DESCRIPTION
**Pull request by udaysagar2177**

### CODE REVIEW

**Comment by udaysagar2177** [comment]
@dpisklov This PR addresses https://github.com/OpenHFT/Chronicle-Map/issues/178.

FYI, due to this PR, net.openhft.chronicle.hash.ChronicleHashBuilder#setPreShutdownAction is little less useful.

File: src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java [hunk 1]
```diff
@@ -1607,6 +1608,12 @@ boolean aligned64BitMemoryOperationsAtomic() {
         return this;
     }
 
+    @Override
+    public ChronicleMapBuilder<K, V> skipCloseOnExitHook() {
+        this.skipCloseOnExitHook = true;
+        return null;
```
Comment (dpisklov) [inline src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java]: why is it returning null?
And why setter method doesn't accept boolean as a parameter?
```suggestion
        return this;
```

Comment (udaysagar2177) [reply src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java]: I thought the method name makes the intention clear. But I can add it if you prefer the parameter.

Comment (udaysagar2177) [reply src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java]: fixed the null return

Comment (dpisklov) [reply src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java]: I'd prefer consistency with other setters - e.g. `B allowSegmentTiering(boolean allowSegmentTiering)` or `B checksumEntries(boolean checksumEntries)`

Comment (udaysagar2177) [reply src/main/java/net/openhft/chronicle/map/ChronicleMapBuilder.java]: Done

**Comment by dpisklov** [comment]
Merged - thanks

**Comment by udaysagar2177** [comment]
@dpisklov Could you please get this change released so that we can use it?

**Comment by RobAustin** [comment]
I will update the chronicle-bom to include the latest snapshot version of chronicle-map, so that the change is included in our next release.

**Comment by dpisklov** [comment]
Release 3.17.3 is out

**Pull request merged**